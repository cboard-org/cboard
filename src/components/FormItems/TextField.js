import React from 'react';
import MUITextField from 'material-ui/TextField';
import { FormControl, FormHelperText } from 'material-ui/Form';

const TextField = ({ className, error, ...props }) => (
  <FormControl className={className} error={!!error}>
    <MUITextField error={!!error} {...props} />
    {error && <FormHelperText>{error}</FormHelperText>}
  </FormControl>
);

export default TextField;
