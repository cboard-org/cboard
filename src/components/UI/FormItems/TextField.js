import React from 'react';
import PropTypes from 'prop-types';
import MUITextField from '@material-ui/core/TextField';
import FormControl from '@material-ui/core/FormControl';
import FormHelperText from '@material-ui/core/FormHelperText';

const propTypes = {
  className: PropTypes.string,
  error: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.string,
    PropTypes.number
  ])
};

const defaultProps = {
  className: '',
  error: false
};

const TextField = ({ className, error, ...props }) => (
  <FormControl className={className} error={!!error}>
    <MUITextField error={!!error} {...props} />
    {error && <FormHelperText>{error}</FormHelperText>}
  </FormControl>
);

TextField.propTypes = propTypes;
TextField.defaultProps = defaultProps;

export default TextField;
