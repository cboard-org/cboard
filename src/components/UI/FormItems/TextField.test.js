import React from 'react';
import { shallowMatchSnapshot } from '../../../common/test_utils';
import TextField from './TextField';

describe('TextField tests', () => {
  test('default renderer', () => {
    shallowMatchSnapshot(<TextField error={'myerror'} />);
  });
});
