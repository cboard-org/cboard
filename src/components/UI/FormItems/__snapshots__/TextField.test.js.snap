// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField tests default renderer 1`] = `
<WithStyles(FormControl)
  className=""
  error={true}
>
  <TextField
    error={true}
    required={false}
    select={false}
    variant="standard"
  />
  <WithStyles(FormHelperText)>
    myerror
  </WithStyles(FormHelperText)>
</WithStyles(FormControl)>
`;
